src/main/java/com/synopsys/integration/blackduck/codelocation/signaturescanner/command/ScanCommand.java

```diff
@@ -169,15 +171,19 @@ public ScanCommand(final File installDirectory, final File outputDirectory, fina
                 if (fullSnippetScan) {
                     cmd.add("--full-snippet-scan");
                 }
+            }
 
-                if (uploadSource) {
-                    cmd.add("--upload-source");
-                }
+            if (licenseSearch) {
+                cmd.add("--license-search");
+            }
+
+            if (uploadSourceProperties.isUploadSource()) {
+                cmd.add("--upload-source");
             }
         } else {
             logger.info("You have configured this signature scan to run in dry run mode - no results will be submitted to Black Duck.");
-            if (snippetMatching || snippetMatchingOnly || fullSnippetScan || uploadSource) {
-                logger.warn("No snippet functionality is supported when running a dry run signature scan.");
+            if (snippetMatching || snippetMatchingOnly || fullSnippetScan || uploadSource || licenseSearch) {
```

Comment:
can this boolean condition be added as part of the UploadSourceProperties?

Reply:
It may be benifitial to expand the UploadSourceProperties class to a class that determines what can and can't be run